metrics/ci_integrations/test/source/github_actions/config/validation_delegate/github_actions_source_validation_delegate_test.dart

```diff
@@ -430,5 +532,346 @@ void main() {
         expect(interactionResult.isSuccess, isTrue);
       },
     );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow runs with conclusion fetching failed",
+      () async {
+        whenFetchWorkflowRunsWithConclusion(workflowId).thenErrorWith();
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
+      },
+    );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow runs with conclusion fetching result is null",
+      () async {
+        whenFetchWorkflowRunsWithConclusion(workflowId).thenSuccessWith(null);
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
+      },
+    );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow runs with conclusion fetching result contains an empty list of workflow runs",
+      () async {
+        whenFetchWorkflowRunsWithConclusion(
+          workflowId,
+        ).thenSuccessWith(emptyRunsPage);
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
+      },
+    );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow run jobs page fetching failed",
+      () async {
+        whenFetchRunJobs().thenErrorWith();
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
+      },
+    );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow run jobs page fetching result is null",
+      () async {
+        whenFetchRunJobs().thenSuccessWith(null);
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
+      },
+    );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the next workflow run jobs page fetching failed",
```

Comment:
Would it be great to test that this method does not fetch the next workflow runs page if the first one contains a job with the given name?